src/chunk_append/chunk_append.c

```diff
@@ -198,13 +203,20 @@ ts_chunk_append_path_create(PlannerInfo *root, RelOptInfo *rel, Hypertable *ht,
 
 			foreach (lc_oid, current_oids)
 			{
-				Assert(lfirst_oid(lc_oid) ==
-					   root->simple_rte_array[((Path *) lfirst(flat))->parent->relid]->relid);
-				merge_childs = lappend(merge_childs, lfirst(flat));
-				flat = lnext(flat);
+				/* postgres may have pruned away some children already */
+				if (lfirst_oid(lc_oid) ==
+					   root->simple_rte_array[((Path *) lfirst(flat))->parent->relid]->relid)
+				{
+					Assert(lfirst_oid(lc_oid) ==
+						root->simple_rte_array[((Path *) lfirst(flat))->parent->relid]->relid);
+					merge_childs = lappend(merge_childs, lfirst(flat));
+					flat = lnext(flat);
+				}
 			}
 
-			if (list_length(merge_childs) > 1)
+			if (merge_childs == NIL)
```

Comment:
yeah, so we could write this as
```C
if (list_length(merge_childs) < 1)
{
/*nop*/
}
else if (list_length(merge_childs) == 1 )
{
...
}
else
{
...
}
```
I held off because it was a larger change.

Reply:
Thanks. Now I see. It wasn't obvious by looking to PR in GH.
May be :
```c
if (list_length(merge_childs) > 1)
{
...
}
else if (list_length(merge_childs) == 1)
{
}
```

It will be less changes in the PR and more obvious to grasp. I guess further refactoring is for a separate issue.